### PR TITLE
added option to choose sort order on column switch between keep same order or reset to asc…

### PR DESCRIPTION
**Pull request by ghost**
## Related Issue
None

## Description

Default web UX still sets sort order to 'asc' when switching between sorted columns. Examples on material-ui table (https://material-ui.com/components/tables/#sorting-amp-selecting) and antd table (https://ant.design/components/table/#components-table-demo-head). As this PR (https://github.com/material-table-core/core/pull/126/commits/a64c4a66681afbb686fc9f84a0ff493c557e48e2) already asked for changing this behaviour, we're now asking for enabling the choice.

## Related PRs
https://github.com/material-table-core/core/pull/126/commits/a64c4a66681afbb686fc9f84a0ff493c557e48e2

## Impacted Areas in Application
Sorting

List general components of the application that this PR will affect:
MTableHeader

### CODE REVIEW

**Comment by matthewoestreich** [comment]
Hello! Thank you for this. I think I understand what you're talking about but just wanted to clarify what you mean by "switching columns"?

**Comment by LeComptoirDesPharmacies** [comment]
> Hello! Thank you for this. I think I understand what you're talking about but just wanted to clarify what you mean by "switching columns"?

Ok I mean when a sort order is set on a column, and you change sorted column.

**Comment by Domino987** [comment]
Hi I think thas a good idea. Can we remove the changes that are not connected to the PR? The icons for example so we can see where code changed in connection to the feature?

File: src/components/MTableHeader/index.js [hunk 1]
```diff
@@ -140,11 +140,18 @@ export function MTableHeader({ onColumnResized, ...props }) {
             <TableSortLabel
               IconComponent={props.icons.SortArrow}
               active={props.orderBy === columnDef.tableData.id}
-              direction={props.orderDirection || 'asc'}
+              direction={
```
Comment (Domino987) [inline src/components/MTableHeader/index.js:170]: Can we combine these tw osections of determining the direction into one variable above the render so we dont have the same logic at several location?

File: src/defaults/props.options.js [hunk 2]
```diff
@@ -40,6 +40,7 @@ export default {
   selection: false,
   selectionProps: {},
   sorting: true,
+  keepSortOrderOnColumnSwitch: true,
```
Comment (Domino987) [inline src/defaults/props.options.js]: Settings the default as true would change the behavior of existing tables right?

File: types/index.d.ts [hunk 3]
```diff
@@ -287,33 +287,42 @@ export const MTableToolbar: (props: any) => React.ReactElement<any>;
 export const MTable: (props: any) => React.ReactElement<any>;
 
 export interface Icons {
```
Comment (Domino987) [inline types/index.d.ts]: Can we remove these changes that are not part of the PR?

**Review by Domino987 — CHANGES_REQUESTED** [review]
This looks like a good idea to do. Can I get some feedback regarding the review questions?

File: src/defaults/props.options.js (hunk 2 above)
Comment (LeComptoirDesPharmacies) [reply src/defaults/props.options.js]: No, it's for being compatible with actual behaviour I set default to true. This means sort order is by default conserved when switching between sorted columns, as it was asked and implemented in this PR: https://github.com/material-table-core/core/commit/a64c4a66681afbb686fc9f84a0ff493c557e48e2

File: types/index.d.ts (hunk 3 above)
Comment (LeComptoirDesPharmacies) [reply types/index.d.ts]: Yes done. Sorry :)

File: src/defaults/props.options.js (hunk 2 above)
Comment (Domino987) [reply src/defaults/props.options.js]: Ok perfect, I just wanted to clarify this. Looks good.

File: src/components/MTableHeader/index.js (hunk 1 above)
Comment (LeComptoirDesPharmacies) [reply src/components/MTableHeader/index.js:170]: Seems complicated. Indeed, they all determine the direction but not at the same moment: what we calculate for "direction" (line 143) is the calculated orderDirection, and what we calculate for orderDirection in the onClick function (line 150) is the future orderDirection. First one is consequence of second one. I did not find a way to calculte them the same way.
I am opened to your suggestions for that. But after spending time on that, I have not found solution.

Comment (Domino987) [reply src/components/MTableHeader/index.js:170]: True, maybe it would help to either create a new function that generates the next or current sorting direction. Also some comments would be needed to explain each step and how the work together so that we can maintain this in the future without figuring out under which condition which string is returned.

**Review by Domino987 — APPROVED** [review]
